pytest_splunk_addon/splunk.py

```diff
@@ -759,13 +760,22 @@ def splunk_ingest_data(request, splunk_hec_uri, sc4s, uf, splunk_events_cleanup)
             )
             sleep(50)
         except Exception as e:
+            with open("ingestion_error_main_worker.txt", "w") as f:
+                f.write(f"{type(e).__name__},{str(e)}")
             raise e
-        finally:
+        else:
             if "PYTEST_XDIST_WORKER" in os.environ:
                 with open(os.environ.get("PYTEST_XDIST_TESTRUNUID") + "_wait", "w+"):
                     PYTEST_XDIST_TESTRUNUID = os.environ.get("PYTEST_XDIST_TESTRUNUID")
     else:
         while not os.path.exists(os.environ.get("PYTEST_XDIST_TESTRUNUID") + "_wait"):
+            if os.path.exists("ingestion_error_main_worker.txt"):
+                with open("ingestion_error_main_worker.txt") as f:
+                    exception_to_throw, message_to_throw = f.readline().split(
```

Comment:
Looks like this `readline()` could happen between opening the file in the main worker (line 763) and flushing its content. In that case, for a short time, the file already exists, but the line read result would be an empty string and attempt to split would raise an exception.